pyro/distributions/affine_beta.py

```diff
@@ -0,0 +1,107 @@
+from torch.distributions.transforms import AffineTransform
+
+
+class AffineBeta(TransformedDistribution):
+    r"""
+    Beta distribution scaled by :attr:`scale` and shifted by :attr:`loc`::
+
+        X ~ Beta(concentration1, concentration0)
+        f(X) = loc + scale * X
+        Y = f(X) ~ AffineBeta(concentration1, concentration0, loc, scale)
+
+    :param float or torch.Tensor concentration1: 1st concentration parameter
+        (alpha) for the Beta distribution.
+    :param float or torch.Tensor concentration0: 2nd concentration parameter
+        (beta) for the Beta distribution.
+    :param float or torch.Tensor loc: location parameter.
+    :param float or torch.Tensor scale: scale parameter.
+    """
+
+    arg_constraints = {
+        "concentration1": constraints.positive,
+        "concentration0": constraints.positive,
+        "loc": constraints.real,
+        "scale": constraints.positive,
+    }
+
+    def __init__(self, concentration1, concentration0, loc, scale, validate_args=None):
+        base_dist = Beta(concentration1, concentration0)
+        super(AffineBeta, self).__init__(
+            base_dist,
+            AffineTransform(loc=loc, scale=scale),
+            validate_args=validate_args,
+        )
+
+    def expand(self, batch_shape, _instance=None):
+        new = self._get_checked_instance(AffineBeta, _instance)
+        return super(AffineBeta, self).expand(batch_shape, _instance=new)
+
+    def sample(self, sample_shape=torch.Size()):
+        """
+        Generates a sample from `Beta` distribution and applies `AffineTransform`.
+        Additionally clamps the output in order to avoid `NaN` and `Inf` values
+        in the gradients.
+        """
+        with torch.no_grad():
+            x = self.base_dist.sample(sample_shape)
+            for transform in self.transforms:
+                x = transform(x)
+            eps = torch.finfo(x.dtype).eps * self.scale
+            x = torch.min(torch.max(x, self.low + eps), self.high - eps)
+            return x
+
+    def rsample(self, sample_shape=torch.Size()):
+        """
+        Generates a sample from `Beta` distribution and applies `AffineTransform`.
+        Additionally clamps the output in order to avoid `NaN` and `Inf` values
+        in the gradients.
+        """
+        x = self.base_dist.rsample(sample_shape)
+        for transform in self.transforms:
+            x = transform(x)
+        eps = torch.finfo(x.dtype).eps * self.scale
+        x = torch.min(torch.max(x, self.low + eps), self.high - eps)
+        return x
+
+    @constraints.dependent_property
+    def support(self):
+        return constraints.interval(self.low, self.high)
+
+    @property
+    def concentration1(self):
+        return self.base_dist.concentration1
+
+    @property
+    def concentration0(self):
+        return self.base_dist.concentration0
+
+    @property
+    def size(self):
+        return self.concentration1 + self.concentration0
```

Comment:
This name of property seems a confusing, since in PyTorch `.size()` is often used as a discrete quantity. Would you mind omitting this one property, maybe defining it in your user code?

Reply:
If I rename it to `.sample_size` like here https://en.wikipedia.org/wiki/Beta_distribution#Mean_and_sample_size would it still be confusing? If that doesn't work either I'll remove it from here and just keep it in my code as you suggested.

Reply:
Sure, `.sample_size` seems ok.